### PR TITLE
Convert tag field to student group

### LINKED ISSUE #90: Convert tag field to student group

### PR DESCRIPTION
**Pull request by teekoksiang**
- Convert tag field to student group
- Update the codes to ensure it did not break the functionalities and tests

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -100,30 +100,30 @@ public static Email parseEmail(String email) throws ParseException {
...
     /**
-     * Parses {@code Collection<String> tags} into a {@code Set<Tag>}.
+     * Parses {@code Collection<String> studyGroups} into a {@code Set<StudyGroup>}.
      */
-    public static Set<Tag> parseTags(Collection<String> tags) throws ParseException {
-        requireNonNull(tags);
-        final Set<Tag> tagSet = new HashSet<>();
-        for (String tagName : tags) {
-            tagSet.add(parseTag(tagName));
+    public static Set<StudentGroup> parseStudyGroups(Collection<String> studyGroups) throws ParseException {
```
Comment (athuyaoo) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java]: sorry StudyGroups and studentGroups are a bit different I feel?

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 2]
```diff
@@ -25,23 +25,25 @@
...
+                getStudentGroupSet("family"), new Block("A"), new Room("405"), new MatriculationNumber("A0123456E")),
             new Person(new Name("Irfan Ibrahim"), new Phone("92492021"), new Email("irfan@example.com"),
                 new Address("Blk 47 Tampines Street 20, #17-35"), new Gender("M"),
-                getTagSet("classmates"), new Block("A"), new Room("405"), new MatriculationNumber("A0123456F")),
+                getStudentGroupSet("classmates"), new Block("A"), new Room("405"),
+                    new MatriculationNumber("A0123456F")),
             new Person(new Name("Roy Balakrishnan"), new Phone("92624417"), new Email("royb@example.com"),
                 new Address("Blk 45 Aljunied Street 85, #11-31"), new Gender("M"),
-                getTagSet("colleagues"), new Block("A"), new Room("405"), new MatriculationNumber("A0123456G"))
+                getStudentGroupSet("colleagues"), new Block("A"), new Room("405"), new MatriculationNumber("A0123456G"))
```
Comment (athuyaoo) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java]: maybe can change these to example student groups like basketball, soccer, choir etc?

File: src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "block" : "A",
     "room" : "405",
     "gender": "F",
-    "tagged": [ "friends" ],
+    "studentGroups": [ "friends" ],
```
Comment (athuyaoo) [inline src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json]: can change studentGroup here to something else like basketball/ soccer

File: src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json [hunk 4]
```diff
@@ -8,8 +8,8 @@
     "block" : "A",
     "room" : "405",
     "gender" : "F",
-    "tagged" : [ "friends" ],
-    "matriculationNumber": "A0123456B"
+    "matriculationNumber": "A0123456B",
+    "studentGroups" : [ "friends" ]
```
Comment (athuyaoo) [inline src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json]: Same here

File: src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json [hunk 5]
```diff
@@ -18,8 +18,8 @@
     "block" : "A",
     "room" : "405",
     "gender" : "M",
-    "tagged" : [ "owesMoney", "friends" ],
-    "matriculationNumber": "A0123456C"
+    "matriculationNumber": "A0123456C",
+    "studentGroups" : [ "owesMoney", "friends" ]
```
Comment (athuyaoo) [inline src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json]: Same here

File: src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json [hunk 6]
```diff
@@ -38,8 +38,8 @@
     "block" : "A",
     "room" : "405",
     "gender" : "M",
-    "tagged" : [ "friends" ],
-    "matriculationNumber": "A0123456E"
+    "matriculationNumber": "A0123456E",
+    "studentGroups" : [ "friends" ]
```
Comment (athuyaoo) [inline src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json]: Same here

**Review by athuyaoo — COMMENTED** [review]
Looks ok in general! Just a few comments regarding naming of method

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (athuyaoo) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: I think it might be more suitable to change this to studentGroups from StudyGroups?

Comment (teekoksiang) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Yes, accidentally typed "studyGroup".

File: src/main/java/seedu/address/model/util/SampleDataUtil.java (hunk 2 above)
Comment (teekoksiang) [reply src/main/java/seedu/address/model/util/SampleDataUtil.java]: Good point, will update it.

**Review by athuyaoo — APPROVED** [review]
LGTM!

**Pull request merged**